File: apps/node/src/fizzbuzz/domain/type/fizz-buzz-type-02.ts
import { FizzBuzzValue } from "../model/fizz-buzz-value.js";
import type { FizzBuzzType } from "./fizz-buzz-type.js";

export class FizzBuzzType02 implements FizzBuzzType {
  generate(number: number): FizzBuzzValue {
    return new FizzBuzzValue(number.toString(), number);
  }
}

File: apps/rust/src/domain/types/fizz_buzz_type02.rs
use crate::domain::model::FizzBuzzValue;

use super::FizzBuzzType;

pub struct FizzBuzzType02;

impl FizzBuzzType for FizzBuzzType02 {
    fn generate(&self, number: i32) -> FizzBuzzValue {
        FizzBuzzValue::new(number, number.to_string())
    }
}
